server/datastore/mysql/migrations/tables/20220713091130_AddOperatingSystemsTable.go

```diff
@@ -0,0 +1,32 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220713091130, Down_20220713091130)
+}
+
+func Up_20220713091130(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE operating_systems (
+    id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+	version VARCHAR(255) NOT NULL,
+	arch VARCHAR(255) NOT NULL,
+	kernel_version VARCHAR(255) NOT NULL
```

Comment:
Should we have a unique constraint on name, version, arch, and kernel_version? We may run in to race conditions if there are multiple inserts for the same operating system.

Reply:
So are you thinking something like:
`CONSTRAINT unique_os UNIQUE (name, version, arch, kernel_version)`?

Reply:
yes, exactly